django/db/migrations/autodetector.py

```diff
@@ -576,6 +583,107 @@ def generate_renamed_models(self):
                             self.old_model_keys.add((app_label, model_name))
                             break
 
+    def generate_moved_models(self):
+        """
+        Find any moved models, generate the operations for them, and remove
+        the old entry from the model lists. Must be run before other
+        model-level generation.
+        """
+        old_keys = self.old_model_keys | self.old_unmanaged_keys
+        added_models = self.new_model_keys - old_keys
+        added_unmanaged_models = self.new_unmanaged_keys - old_keys
+        all_added_models = chain(
+            sorted(added_models, key=self.swappable_first_key),
+            sorted(added_unmanaged_models, key=self.swappable_first_key),
+        )
+        new_keys = self.new_model_keys | self.new_unmanaged_keys
+        deleted_models = self.old_model_keys - new_keys
+        deleted_unmanaged_models = self.old_unmanaged_keys - new_keys
+        all_deleted_models = chain(
+            sorted(deleted_models), sorted(deleted_unmanaged_models)
+        )
+        for app_label, model_name in all_added_models:
+            model_state = self.to_state.models[app_label, model_name]
+            model_fields_def = self.only_relation_agnostic_fields(model_state.fields)
+            for rem_app_label, rem_model_name in all_deleted_models:
+                if rem_app_label == app_label:
+                    continue
+                rem_model_state = self.from_state.models[rem_app_label, rem_model_name]
+                rem_model_fields_def = self.only_relation_agnostic_fields(
+                    rem_model_state.fields
+                )
+                if (
+                    model_fields_def == rem_model_fields_def
+                    and not self.questioner.ask_move_model(
+                        rem_model_state,
+                        model_state,
+                    )
+                ):
+                    continue
+                # Quit if db_table is not explicitly provided in moved model.
+                if not model_state.options.get(
+                    "db_table",
+                ):
+                    self.questioner.quit_for_db_table_add(
+                        model_name, f"{rem_app_label}_{model_name}"
+                    )
```

Comment:
Could we possibly continue here and simply include the `RenameModel` operation as well?